packages/store-api/src/platforms/vtex/clients/search/index.ts

```diff
@@ -37,13 +33,16 @@ export const IntelligentSearch = (opts: Options) => {
   const { channel } = opts
   const base = `http://search.biggylabs.com.br/search-api/v1/${opts.account}`
 
-  // TODO: change here once supporting sales channel
-  const defaultFacets = [
-    {
-      key: 'trade-policy',
-      value: channel,
-    },
-  ]
+  const addDefaults = (facets: SelectedFacet[]) => {
+    const facetsObj = Object.fromEntries(
+      facets.map(({ key, value }) => [key, value])
+    )
+
+    return Object.entries({
+      'trade-policy': channel,
+      ...facetsObj,
+    }).map(([key, value]) => ({ key, value }))
```

Comment:
What do you think about this suggestion? Have an if statement may not harm the readability.

```ts
const tradePolicy = facets.find(({ key }) => key === 'trade-policy')

if (tradePolicy) {
  return facets
}

return facets.concat([{ 'trade-policy': channel }]) // can do [...facets, { 'trade-policy': channel }]
```
This code, for me, seems easier to read than the other.

Reply:
I agree with @igorbrasileiro. It may be slightly less performant because we don't perform a search, but this array is usually so small that I don't think it'll matter.

Just a correction, the last line should be:
```tsx
return facets.concat([{ key: 'trade-policy', value: channel }]) // can do [...facets, { key: 'trade-policy', value: channel }]
```

Reply:
I agree for this case, but what if we need to add a second parameter? Wouldn't it make a spaghetti code?

Reply:
You have a point; ok! I agree with you